shelley/chain-and-ledger/shelley-spec-ledger-test/src/Test/Shelley/Spec/Ledger/Example.hs

```diff
@@ -0,0 +1,252 @@
+{-# LANGUAGE DataKinds #-}
+{-# LANGUAGE EmptyDataDecls #-}
+{-# LANGUAGE DefaultSignatures #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE DeriveTraversable #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE FlexibleInstances #-}
+{-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Test.Shelley.Spec.Ledger.Example where
+
+import Control.Monad
+import Control.State.Transition.Extended
+import Data.Foldable
+import Data.Functor.Compose
+import Data.Group
+import Data.Map (Map)
+import Data.Monoid
+import Data.Set (Set)
+import Data.Typeable
+import qualified Data.Map as Map
+import qualified Data.Set as Set
+import Data.Void
+import Data.Default
+
+import Shelley.Spec.Ledger.API.Validation
+import qualified Cardano.Ledger.Core as Core
+
+
+-- TODO: find this or put this in a library
+newtype MonoidMap k w = MonoidMap { getMonoidMap :: Map k w }
+  deriving (Eq, Show, Foldable)
+
+instance (Ord k, Monoid w, Eq w) => Semigroup (MonoidMap k w) where
+  MonoidMap xs <> MonoidMap ys = MonoidMap $ Map.filter (/= mempty) $ Map.unionWith (<>) xs ys
+
+instance (Ord k, Monoid w, Eq w) => Monoid (MonoidMap k w) where
+  mempty = MonoidMap Map.empty
+
+instance (Ord k, Group w, Eq w) => Group (MonoidMap k w) where
+  invert (MonoidMap xs) = MonoidMap (invert <$> xs)
+
+
+-- TODO: find this or put this in a library
+data Valid e a = Invalid e | Valid a
+  deriving (Functor, Foldable, Traversable, Eq, Ord, Show)
+
+valid :: (e -> c) -> (a -> c) -> Valid e a -> c
+valid e2c a2c = \case
+  Invalid e -> e2c e
+  Valid a -> a2c a
+
+instance Semigroup e => Applicative (Valid e) where
+  pure = Valid
+  Invalid l <*> Invalid r = Invalid (l <> r)
+  Invalid l <*> Valid _ = Invalid l
+  Valid _ <*> Invalid r = Invalid r
+  Valid f <*> Valid x = Valid (f x)
+
+deriving via (Ap (Valid e) a)
+  instance (Semigroup e, Semigroup a) => Semigroup (Valid e a)
+
+deriving via (Ap (Valid e) a)
+  instance (Semigroup e, Monoid a) => Monoid (Valid e a)
+
+
+type family MOutput a
+type family MTxId a
+type family MInvalidTxn a
+
+data ModelChain a
+
+data ModelChainPredicateFailure a
```

Comment:
I don't understand what this is and why it's living in `Example`

Reply:
I never intended to commit this in the first place and have deleted it.